### PR TITLE
Develop a Base for Bar Chart #1

### PR DESCRIPTION
**Pull request by Gayathri-GA**
**Changes done** #1 
Added responsive axes change w.r.t vertical.
Added computeDryLayout function based on box constraints for performing layouts .
Created class for rendering labels, axes and rulers views for the bar chart with initial implementation.

### CODE REVIEW

File: lib/src/bar_chart/bar_chart_axes.dart [hunk 1]
```diff
@@ -1,46 +1,186 @@
...
+  ///
+  /// The [numAxisPoints], [barChartHeight], and [barChartWidth] parameters are
+  /// optional. If not provided, they default to 10, 600, and 1500, respectively.
   const BarChartAxes({
-    super.key,
-    required this.value,
-  });
+    Key? key,
+    this.numAxisPoints = 10,
+    this.barChartHeight = 600,
```
Comment (hasnentai) [inline lib/src/bar_chart/bar_chart_axes.dart]: Can we remove this parameter both height and width?

File: lib/src/bar_chart/bar_chart_label.dart [hunk 2]
```diff
@@ -1,46 +1,243 @@
...
+  /// The [barChartHeight] parameter specifies the height of the entire bar chart.
 
-  const BarChartLabel({
-    super.key,
-    required this.value,
-  });
+  const BarChartLabel(
+      {Key? key,
+      this.numAxisPoints = 10,
+      this.labelTextStyle = const TextStyle(color: Colors.black, fontSize: 12),
```
Comment (hasnentai) [inline lib/src/bar_chart/bar_chart_label.dart]: can we rename `labelTextStyle` to `textStyle`?

File: lib/src/bar_chart/bar_chart_label.dart [hunk 3]
```diff
@@ -1,46 +1,243 @@
...
+      this.barChartHeight = 600,
```
Comment (hasnentai) [inline lib/src/bar_chart/bar_chart_label.dart]: I dont think we need height and width over here as well.
whats your thought?

File: lib/src/bar_chart/bar_chart_rulers.dart [hunk 4]
```diff
@@ -1,46 +1,177 @@
...
+      markNeedsLayout();
+    }
+  }
+
+  /// Creates a [RenderBarChartRulers] object.
+  ///
+  /// The [numAxisPoints], [barChartWidth], and [barChartHeight] arguments
+  /// are required.
+  RenderBarChartRulers({
+    required int numAxisPoints,
```
Comment (hasnentai) [inline lib/src/bar_chart/bar_chart_rulers.dart]: I need to know this `numAxisPoints` what its exactly used for

**Review by hasnentai — CHANGES_REQUESTED** [review]

File: lib/src/bar_chart/bar_chart_label.dart (hunk 2 above)
Comment (Gayathri-GA) [reply lib/src/bar_chart/bar_chart_label.dart]: @hasnentai Yes, that could be done.
Resolved and updated code on this [commit.](https://github.com/GeekyAnts/geekyants_flutter_charts/commit/794f4142a855a03b4e7828c6a66cd259ed54dc3a)

File: lib/src/bar_chart/bar_chart_rulers.dart (hunk 4 above)
Comment (Gayathri-GA) [reply lib/src/bar_chart/bar_chart_rulers.dart]: @hasnentai [Resolved.](https://github.com/GeekyAnts/geekyants_flutter_charts/commit/794f4142a855a03b4e7828c6a66cd259ed54dc3a)

File: lib/src/bar_chart/bar_chart_axes.dart (hunk 1 above)
Comment (Gayathri-GA) [reply lib/src/bar_chart/bar_chart_axes.dart]: @hasnentai Removed and [Updated](https://github.com/GeekyAnts/geekyants_flutter_charts/commit/794f4142a855a03b4e7828c6a66cd259ed54dc3a).

**Pull request merged**